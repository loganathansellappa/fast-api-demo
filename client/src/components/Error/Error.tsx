export const Error = () => {
  const reloadPage = () => {
    window.location.reload();
  };

  return (
    <div className="flex flex-col items-center justify-center h-screen">
      <h1 className="text-2xl font-bold mb-4">Something went wrong!</h1>
      <button onClick={reloadPage} className="py-2 px-4 bg-blue-500 text-white rounded hover:bg-blue-600 focus:outline-none">
        Reload Page
      </button>
    </div>
  );
};
