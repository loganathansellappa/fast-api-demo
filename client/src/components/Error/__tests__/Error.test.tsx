import { render } from "@testing-library/react";
import { Error } from '../Error.tsx';

let element: { container: HTMLElement };

test("renders Loader component", () => {
  element = render(<Error />);
  expect(element.container.firstChild).toMatchSnapshot();
  expect(element.container.textContent).toContain("Something went wrong!");
});