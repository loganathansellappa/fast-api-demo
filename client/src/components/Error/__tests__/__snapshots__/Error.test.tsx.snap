// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loader component 1`] = `
<div
  class="flex flex-col items-center justify-center h-screen"
>
  <h1
    class="text-2xl font-bold mb-4"
  >
    Something went wrong!
  </h1>
  <button
    class="py-2 px-4 bg-blue-500 text-white rounded hover:bg-blue-600 focus:outline-none"
  >
    Reload Page
  </button>
</div>
`;
